### PR TITLE
Update rpk for v23.2

### LINKED ISSUE #1608: Document `rpk profile` and `rpk cloud auth`, `-X` flag new commands for v23.2
### Describe the Issue

We are introducing a few new features in rpk in v23.2.
* a global `-X` flag to setting configuration
* a new command space, `rpk profile`
* a new command space, `rpk cloud auth`
* changes in behavior the existing command `rpk cloud login`

There are a lot of changes, so we should work tightly to figure out what all needs to be documented. I can try to expand on detail in this ticket, but there's a lot of detail, so it may be worth a quick sync first to break down how we'd like to attack this.

* Pull requests introducing the functionality: https://github.com/redpanda-data/redpanda/pull/9981 and https://github.com/redpanda-data/redpanda/pull/10528
* SMEs: @twmb, @r-vasquez 
* Channel: #discuss-devex
* Included in release v23.2

### PR DESCRIPTION
**Pull request by shorgi**
SME review deadline: Thu Jul 6
- Do validate all rpk commands have reference pages, and call out any content gaps.
- Don't yet review the markdown formatting of rpk command pages (it'll be completed after the content is reviewed)

Fixes #1608 

Previews:
- New heading for [`-X` common configuration options](https://deploy-preview-1612--redpanda-documentation.netlify.app/docs/beta/get-started/intro-to-rpk/#common-configuration-options)
- New page for [rpk Profiles](https://deploy-preview-1612--redpanda-documentation.netlify.app/docs/beta/get-started/intro-to-rpk/config-rpk-profile)
- Updated [rpk command reference pages](https://deploy-preview-1612--redpanda-documentation.netlify.app/docs/beta/reference/rpk), including [`rpk -X options`](https://deploy-preview-1612--redpanda-documentation.netlify.app/docs/beta/reference/rpk/rpk-x-options) and [`rpk generate app`](https://deploy-preview-1612--redpanda-documentation.netlify.app/docs/beta/reference/rpk/rpk-generate/rpk-generate-app)

### CODE REVIEW

File: versioned_docs/version-23.2/get-started/broker-admin.mdx [hunk 1]
```diff
@@ -10,48 +10,52 @@ title: Broker and Admin API Addresses
...
+Previous versions of `rpk` supported the `--brokers` and `--api-urls` flags for broker and Admin API addresses, respectively. Those flags are deprecated in this version. 
+:::
+
+The following table shows which `rpk` commands require the `-X brokers` flag, which require the `-X admin.hosts` or `–-hosts` flag, and which don’t require either flag.
 
 | Command | Address flag required |
 | :------- | -------------------- |
-| `rpk acl` (all except `rpk acl user`) | `--brokers` |
-| `rpk acl user` | `--api-urls` |
+| `rpk acl` (all except `rpk acl user`) | `-X brokers` |
```
Comment (twmb) [inline versioned_docs/version-23.2/get-started/broker-admin.mdx:37]: Should this file include an `=` after the brokers / hosts bit, to show the flag values are incomplete?

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+</head>
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+Use rpk profiles to simplify your development experience using `rpk` with multiple Redpanda clusters by saving and reusing configurations for different clusters.
+
+## About rpk profiles
+
+An rpk profile contains a reusable configuration for a Redpanda cluster. When running `rpk`, you can create a profile, configure it for a cluster you're working with, save it, then use it when running an rpk command for the cluster.
```
Comment (twmb) [inline versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:18]: `save it` -- I think this is implied by `create` and `configure`.

wdyt of:
An rpk profile contains a reusable configuration for a Redpanda cluster. When running `rpk`, you can create a profile, configure it for a cluster you're working with, and then repeatably use it when running an rpk command for the cluster.

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+Use rpk profiles to simplify your development experience using `rpk` with multiple Redpanda clusters by saving and reusing configurations for different clusters.
+
+## About rpk profiles
+
+An rpk profile contains a reusable configuration for a Redpanda cluster. When running `rpk`, you can create a profile, configure it for a cluster you're working with, save it, then use it when running an rpk command for the cluster. 
+
+You can create different profiles for different Redpanda clusters. For example, your local cluster, development cluster, and production cluster can each have their own profile, with all of their information stored in a single, local file. You set a unique name for each profile.
```
Comment (twmb) [inline versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:20]: `single, local file` is too specific, I try to hide this implementation detail in rpk (what if we need more files later?)
wdyt of: "with all of their information managed locally by rpk" ?

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+A profile saves the Redpanda node and cluster properties that rpk can already configure. For details, see [Specifying command properties](../intro-to-rpk#specifying-command-properties).
```
Comment (twmb) [inline versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:22]: Not node and cluster properties, just rpk-specific-command properties.

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+Most `rpk` commands can read configuration values from a profile. You pass a profile to an `rpk` command by setting the `--profile` flag. For example, the command `rpk topic produce dev-topic --profile dev` gets its configuration from the profile named `dev`.
+
+## Work with rpk profiles
+
+The primary tasks for working with rpk profiles:
+
+- Create one or more profiles.
+- Choose the in-use profile.
```
Comment (twmb) [inline versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:31]: `Choose the profile to use`?

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+### Create profile
+
+To create a new profile, run [`rpk profile create`](../../reference/rpk/rpk-profile/rpk-profile-create):
+
+```bash
+rpk profile create <profile-name> [flags]
+```
+
+If no profile name is provided, the name `default` is used. After the profile is created, rpk switches to the newly created profile.
```
Comment (twmb) [inline versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:44]: I removed the `default` name in the latest commit -- it was actually pretty confusing. Instead, a name is required _unless_ the person uses `--from-cloud`, where we will generate the name from the cloud cluster name (if no name is provided at the cli)

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+An rpk profile can be generated from different sources:
+- A `redpanda.yaml` file, using the `--from-redpanda` flag.
+- A different rpk profile, using the `--from-profile` flag.
+- A Redpanda Cloud cluster, using the `--from-cloud` flag.
+
+Configuration can be set during creation with the `--set [key=value]` flag. To simplify configuration, the `--set` flag supports autocompletion of valid keys, suggesting key names based on their `-X` format. 
+
+:::important
```
Comment (twmb) [inline versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:53]: maybe not _so_ important to have a full different CSS block for it, but this is worth nothing.

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+You should always use and set the `--description` flag to describe your profiles. The description is printed in the output of [`rpk profile list`](../../reference/rpk/rpk-profile/rpk-profile-list).
+:::
+
+Created profiles are stored in an `rpk.yaml` file. By default, this file is created as `~/.config/rpk/rpk.yaml`. All profiles created by a developer are stored in the same `rpk.yaml` file.
```
Comment (twmb) [inline versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:57]: It is `~/Library/Application Support/rpk/` on Mac, and somewhere else on windows -- i.e. the OS localized config dir.

File: versioned_docs/version-23.2/get-started/broker-admin.mdx (hunk 1 above)
Comment (shorgi) [reply versioned_docs/version-23.2/get-started/broker-admin.mdx:37]: Will add `=<value>`

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+## About rpk profiles
+
+An rpk profile contains a reusable configuration for a Redpanda cluster. When running `rpk`, you can create a profile, configure it for a cluster you're working with, save it, then use it when running an rpk command for the cluster. 
+
+You can create different profiles for different Redpanda clusters. For example, your local cluster, development cluster, and production cluster can each have their own profile, with all of their information stored in a single, local file. You set a unique name for each profile.
+
+A profile saves the Redpanda node and cluster properties that rpk can already configure. For details, see [Specifying command properties](../intro-to-rpk#specifying-command-properties).
+
+Most `rpk` commands can read configuration values from a profile. You pass a profile to an `rpk` command by setting the `--profile` flag. For example, the command `rpk topic produce dev-topic --profile dev` gets its configuration from the profile named `dev`.
```
[the remark this answers is not in the document]
Comment (shorgi) [reply versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:24]: My misunderstanding, thought I saw a command w/o `--profile` but they all actually do.

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx (hunk 8 above)
Comment (r-vasquez) [reply versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:57]: %AppData% For Windows, but we don't really document the Windows usage.

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx (hunk 7 above)
Comment (shorgi) [reply versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:53]: Changing to `:::note`

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx (hunk 6 above)
Comment (shorgi) [reply versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:44]: Will fix

File: versioned_docs/version-23.2/reference/rpk/rpk-profile/rpk-profile-set.mdx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+</head>
+
+Set fields in the current rpk profile.
+
+As in the create command, this command takes a list of `key=value` pairs to write
+to the current profile.
+
+The key can either be the name of a `-X` flag or the path to the field in the
+profile's yaml format. For example, using `--set tls.enabled=true` or `--set
+kafka_api.tls.enabled=true` is equivalent. The former corresponds to the `-X` flag
```
Comment (ivpanda) [inline versioned_docs/version-23.2/reference/rpk/rpk-profile/rpk-profile-set.mdx:19]: Nit: in the preview, there's an unnecessary line break here, and it looks a little strange. Not sure why: 

image

**Comment by ivpanda** [comment]
Thanks for writing all this! Not sure if this is in scope, but I would appreciate more examples here, if possible? I understand that not a lot of extra detail may be added or desired, but a few examples: 

1. In the common config properties for `-X`, it would be nice to see example (truncated) outputs of either available options, or an example of the detailed descriptions of the options. It also looks somewhat like the common configuration options are deprecated, given the proximity of the text to the blue accordion: 
image
2. When showing how an rpk profile can be created, we have three options, but an example would be useful. For example, we say it can be done using "A redpanda.yaml file, using the --from-redpanda flag," but a contextual example would be great there— stating exactly how the --from-redpanda flag is used in `rpk profile create`.
3. It would be nice to have the keys shown on the `rpk profile set` page. We show the usage guidelines and the flags, but not the keys? A user has to navigate elsewhere to find them, then come back here to see usage and flags? image

This is just UX feedback, feel free to take or leave.

File: versioned_docs/version-23.2/reference/rpk/rpk-profile/rpk-profile-set.mdx (hunk 10 above)
Comment (shorgi) [reply versioned_docs/version-23.2/reference/rpk/rpk-profile/rpk-profile-set.mdx:19]: Good catch. I can't find a way to add a non-breaking space within inline-code, so will have these on their own lines.

**Review by twmb — APPROVED** [review]

File: versioned_docs/version-23.2/get-started/broker-admin.mdx [hunk 11]
```diff
@@ -10,48 +10,52 @@ title: Broker and Admin API Addresses
 
 In order for `rpk` to know where to run Kafka-related commands or admin-related commands, you must provide the broker addresses or admin server addresses for each node of a Redpanda cluster. You can specify these addresses as IP addresses or as hostnames, using any of these methods:
 
-* Command line flag (`--brokers` or `--api-urls`)
+* Command line flag (`-X brokers` or `-X admin.hosts`)
 * Environment variable setting (`REDPANDA_BROKERS` or `REDPANDA_API_ADMIN_ADDRS`)
```
Comment (r-vasquez) [inline versioned_docs/version-23.2/get-started/broker-admin.mdx:14]: Should we start documenting the new env variables? (RPK_BROKERS and RPK_ADMIN_HOSTS) 

cc: @twmb

**Review by r-vasquez — COMMENTED** [review]
I'm +1 Thanks for this @shorgi. My only question is about documenting the new env vars. Not sure if this can be a follow-up PR or should we add it here (it could be a lot)

File: versioned_docs/version-23.2/get-started/broker-admin.mdx (hunk 11 above)
Comment (shorgi) [reply versioned_docs/version-23.2/get-started/broker-admin.mdx:14]: Good catch, fixing this now ...

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+An rpk profile can be generated from different sources:
+- A `redpanda.yaml` file, using the `--from-redpanda` flag.
+- A different rpk profile, using the `--from-profile` flag.
+- A Redpanda Cloud cluster, using the `--from-cloud` flag.
+
+:::note
+You must provide a profile name when creating a profile that isn't generated from a Redpanda Cloud cluster with the `--from-cloud` flag.
+:::
+ 
+After the profile is created, rpk switches to the newly created profile.
```
Comment (Feediver1) [inline versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:53]:
```suggestion
After the profile is created, rpk switches to the newly-created profile.
```

File: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+
+Configuration can be set during creation with the `--set [key=value]` flag. To simplify configuration, the `--set` flag supports autocompletion of valid keys, suggesting key names based on their `-X` format.
```
Comment (Feediver1) [inline versioned_docs/version-23.2/get-started/config-rpk-profile.mdx:55]:
```suggestion
You can specify the configuration during creation with the `--set [key=value]` flag. To simplify configuration, the `--set` flag supports autocompletion of valid keys, suggesting key names based on their `-X` format. 
```

File: versioned_docs/version-23.2/reference/rpk/rpk-x-options.mdx [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+Every `-X` option can be translated into an environment variable by prefixing with `RPK_` and replacing periods (`.`) with underscores (`_`). For example, the flag `tls.enabled` has the equivalent environment variable `RPK_TLS_ENABLED`.
+
+:::important
+- Flags common across all rpk commands in previous versions (for example, `--brokers`, `--tls-enabled`) are deprecated. 
+- Functionality of all deprecated flags are supported as `-X` options.
+:::
+
+:::tip Tips
+- rpk supports command-line (tab) completion for `-X` flag keys.
+- Each rpk command's `-help` text prints information specific to the command. To get description about `-X` options, run `rpk -X list` to list supported options, or run `rpk -X help` to get details about supported options.
```
Comment (Feediver1) [inline versioned_docs/version-23.2/reference/rpk/rpk-x-options.mdx:25]:
```suggestion
- Each rpk command's `-help` text prints information specific to the command. To get a description of `-X` options, run `rpk -X list` to list supported options, or run `rpk -X help` to get details about supported options.
```

File: versioned_docs/version-23.2/reference/rpk/rpk-x-options.mdx [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+
+**Example**: `admin.hosts=localhost:9644,rp.example.com:9644`
+
+---
+
+### admin.tls.enabled
+
+A boolean that enables rpk to speak TLS to your broker's Admin API listeners.
+  
+You can use this if you have well known certificates setup on your Admin API. If you use mTLS, specifying mTLS certificate filepaths automatically opts into `admin.tls.enabled`.
```
Comment (Feediver1) [inline versioned_docs/version-23.2/reference/rpk/rpk-x-options.mdx:116]:
```suggestion
You can use this if you have well known certificates set up on your Admin API. If you use mTLS, specifying mTLS certificate filepaths automatically opts into `admin.tls.enabled`.
```

**Review by Feediver1 — APPROVED** [review]
Left some suggestions here

**Pull request merged**